cms/djangoapps/maintenance/tests.py

```diff
@@ -0,0 +1,243 @@
+        super(TestForcePublish, self).setUp()
+        self.view_url = reverse('maintenance:force_publish_course')
+        # Add some changes to course
+        chapter = ItemFactory.create(category='chapter', parent_location=self.course.location)
+        self.store.create_child(
+            self.user.id,  # pylint: disable=no-member
+            chapter.location,
+            'html',
+            block_id='html_component'
+        )
+        # verify that course has changes.
+        self.assertTrue(self.store.has_changes(self.store.get_item(self.course.location)))
+
+    @ddt.data(
+        ('', COURSE_KEY_ERROR_MESSAGES['empty_course_key']),
+        ('edx', COURSE_KEY_ERROR_MESSAGES['invalid_course_key']),
+        ('course-v1:e+d+X', COURSE_KEY_ERROR_MESSAGES['course_key_not_found']),
+    )
+    @ddt.unpack
+    def test_invalid_course_key_messages(self, course_key, error_message):
+        """
+        Test all error messages for invalid course keys.
+        """
+        # validate that course key contains error message
+        self.verify_error_message(
+            {
+                'course-id': course_key
+            },
+            error_message
+        )
+
+    def test_non_split_course(self):
+        """
+        Test that we get a error message on non-split courses.
+        """
+        # validate non split error message
+        course = CourseFactory.create(default_store=ModuleStoreEnum.Type.mongo)
+        self.verify_error_message(
+            {
+                'course-id': unicode(course.id)
+            },
+            'Force publish course does not support old mongo style courses.'
+        )
+
+    def test_already_published(self):
+        """
+        Test that when a course is forcefully publish, we get a 'course is already published' message.
+        """
+        # force publish the course
+        response = self.get_force_publish_course_response(is_dry_run=False)
+        self.validate_success_from_response(response, 'Forced published the course')
+
+        # now course is forcefully published, we should get already published course.
+        self.verify_error_message(
+            {
+                'course-id': unicode(self.course.id),
+                'dry-run': ''
+            },
+            'Course is already in published state.'
+        )
+
+    def verify_versions_are_different(self):
+        """
+        Verify draft and published versions point to different locations.
+        """
+        # get draft and publish branch versions
+        versions = get_course_versions(unicode(self.course.id))
+
+        # verify that draft and publish point to different versions
+        self.assertNotEqual(versions['draft-branch'], versions['published-branch'])
+
+    def get_force_publish_course_response(self, is_dry_run=True):
+        """
+        Get force publish the course response.
+
+        Argument:
+
+            is_dry_run - default True, that means by default the view does dry run.
+        """
```

Comment:
Can we use pep8 Docstring Conventions? I am open. 
https://www.python.org/dev/peps/pep-0257/